src/map.cpp

```diff
@@ -4259,11 +4259,37 @@ unsigned int temp_difference_ratio( int temp_one, int temp_two )
     return ratio;
 }
 
+int check_freezing_temperature( item &it )
```

Comment:
Wouldn't it be better to make this a per-item .json entry check instead of a hardcoded list of items?

Reply:
Few factors convince me to not go this way. One is relative short list of items with freezing point different then water's. Second is my present lack of knowledge of json handling code. I can and will learn it eventualy but that would postpone this PR and its needed now. And nothing stands in a way of jsonizing it later.  Third is alcohol items would need to be defined one by one while this uses actions to grab them into groups. Fourth is in question form: does this realy need to be modable? (perhaps if a mod adds a new item yes).

Reply:
Right on.

An additional thought: it could also be based off of materials.

Reply:
It can be json based if we have more abstracts for consumables, IIRC you can leave some default value and item without FP will fallback to it.

Reply:
I totally agree with both @Rivet-the-Zombie's points.

We avoid hardcoded items wherever possible in the code (which is littered with them nonetheless). It's basically a mess. People say they'll clean it up later, but most of them never really do. So the mess grows.

JSON interaction code is fairly simple. You could look at `islot_comestible` (appears to be a candidate for storing freezing temperature) and its serialization for an example. In the absence of needful values in `COMESTIBLE`, you can fallback to the material's defaults.

Reply:
For those few items its an overkill nevertheless in my opinion. If I manage to try before my vacation break I will do it, if not this will stall for some time.

Reply:
> For those few items its an overkill nevertheless in my opinion. 

It's probably not. Even for a single item it is worth doing. Hardcoded items tie the core game to this very item (something we would like to avoid). Not only this prevents reliable modding (btw core data considered a mod). Even without modding, bugs and also chunks of dead code will be inevitable later on (when someone adds more items / removes the existing).

Reply:
Ok, I get the idea now.  I'll try to do it. Thanks for pointing me in the right direction.